### ISSUE TITLE
Fail to build. error: control reaches end of non-void function

### ISSUE THREAD

**Issue by kevinsmia1939** [sunqm/libcint#59]
Hi, 
I failed to build this on openSUSE Tumbleweed.
openSUSE enforce this warning flag as error.

```
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ip1_spinor':
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:387:1: error: control reaches end of non-void function [-Werror=return-type]
[   20s]   387 | }
[   20s]       | ^
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ip2_spinor':
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:464:1: error: control reaches end of non-void function [-Werror=return-type]
[   20s]   464 | }
[   20s]       | ^
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ipip1_spinor':
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:1289:1: error: control reaches end of non-void function [-Werror=return-type]
[   20s]  1289 | }
[   20s]       | ^
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ip1ip2_spinor':
[   20s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:1366:1: error: control reaches end of non-void function [-Werror=return-type]
[   20s]  1366 | }
[   20s]       | ^
[   20s] cc1: some warnings being treated as errors
[   20s] make[2]: *** [CMakeFiles/cint.dir/build.make:555: CMakeFiles/cint.dir/src/autocode/int3c2e.c.o] Error 1
[   20s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/libcint-4.4.1/build'
[   20s] make[2]: *** Waiting for unfinished jobs....
```

**Comment by hebrewsnabla** [comment]
Which C compiler do you use? I didn't meet that error with gcc10 or gcc11.

**Comment by kevinsmia1939** [comment]
> Which C compiler do you use? I didn't meet that error with gcc10 or gcc11.

gcc11
Werror=return-type is use, did it just appear as warning for you?

**Comment by hebrewsnabla** [comment]
> > Which C compiler do you use? I didn't meet that error with gcc10 or gcc11.
> 
> gcc11
> Werror=return-type is use, did it just appear as warning for you?

I didn't even meet any warning. I've tried simple `cmake ..` and `cmake -DWITH_F12=1 -DWITH_RANGE_COULOMB=1 -DWITH_FORTRAN=1 -DWITH_CINT2_INTERFACE=1 -DMIN_EXPCUTOFF=20 ..` with gcc11.1.0 on Archlinux, and everything looks ok.

**Comment by kevinsmia1939** [comment]
Hi,
I tried adding `return 1;` to the line reported above and it now build just fine.

I will look into what is causing this error.
Here is the build flag.
```
[   21s] [ 55%] Building C object CMakeFiles/cint.dir/src/autocode/int3c2e.c.o
[   21s] /usr/bin/cc -DWITH_CINT2_INTERFACE -DWITH_COULOMB_ERF -DWITH_F12 -DWITH_FORTRAN -Dcint_EXPORTS -I/home/abuild/rpmbuild/BUILD/libcint-4.4.1/build/include -I/home/abuild/rpmbuild/BUILD/libcint-4.4.1/src -I/home/abuild/rpmbuild/BUILD/libcint-4.4.1/build/src -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -O2 -g -DNDEBUG -fPIC -MD -MT CMakeFiles/cint.dir/src/autocode/int3c2e.c.o -MF CMakeFiles/cint.dir/src/autocode/int3c2e.c.o.d -o CMakeFiles/cint.dir/src/autocode/int3c2e.c.o -c /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ip1_spinor':
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:387:1: error: control reaches end of non-void function [-Werror=return-type]
[   21s]   387 | }
[   21s]       | ^
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ip2_spinor':
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:464:1: error: control reaches end of non-void function [-Werror=return-type]
[   21s]   464 | }
[   21s]       | ^
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ipip1_spinor':
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:1289:1: error: control reaches end of non-void function [-Werror=return-type]
[   21s]  1289 | }
[   21s]       | ^
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c: In function 'int2c2e_ip1ip2_spinor':
[   21s] /home/abuild/rpmbuild/BUILD/libcint-4.4.1/src/autocode/int3c2e.c:1366:1: error: control reaches end of non-void function [-Werror=return-type]
[   21s]  1366 | }
[   21s]       | ^
[   21s] cc1: some warnings being treated as errors
[   21s] make[2]: *** [CMakeFiles/cint.dir/build.make:555: CMakeFiles/cint.dir/src/autocode/int3c2e.c.o] Error 1
```

**Comment by sunqm** [comment]
Bug fixed in 4632807

**Comment by kevinsmia1939** [comment]
thank so much for the help
